ios/BT/ExposureManager.swift

```diff
@@ -297,7 +301,72 @@ final class ExposureManager: NSObject {
     }
     return progress
   }
-  
+
+  @available(iOS 13.7, *)
+  @discardableResult func detectExposuresV2(completionHandler: @escaping ((ExposureResult) -> Void)) -> Progress {
+
+    let progress = Progress()
+    var lastProcessedUrlPath: String = .default
+    var processedFileCount: Int = 0
+    var unpackedArchiveURLs: [URL] = []
+
+    Promise<[Exposure]>(on: .global()) { () -> [Exposure] in
+      if self.isDetectingExposures {
+        // Disallow concurrent exposure detection,
+        // because if allowed we might try to detect the same diagnosis keys more than once
+        throw ExposureError.default("Detection Already in Progress")
+      }
+      self.isDetectingExposures = true
+      // Reset file capacity to 15 if > 24 hours have elapsed since last reset
+      self.updateRemainingFileCapacity()
+      guard self.btSecureStorage.userState.remainingDailyFileProcessingCapacity > 0 else {
+        // Abort if daily file capacity is exceeded
+        return []
+      }
+      let indexFileString = try await(self.fetchIndexFile())
+      let remoteURLs = indexFileString.gaenFilePaths
+      let targetUrls = self.urlPathsToProcess(remoteURLs)
+      lastProcessedUrlPath = targetUrls.last ?? .default
+      processedFileCount = targetUrls.count
+      let downloadedKeyArchives = try await(self.downloadKeyArchives(targetUrls: targetUrls))
+      unpackedArchiveURLs = try await(self.unpackKeyArchives(packages: downloadedKeyArchives))
+      let exposureConfiguraton = try await(self.getExposureConfigurationV2())
+      var exposureSummary = try await(self.callDetectExposures(configuration: exposureConfiguraton.asENExposureConfiguration,
+                                                               diagnosisKeyURLs: unpackedArchiveURLs))
+      exposureSummary = try await(self.callAggregateDetectExposures(configuration: exposureConfiguraton.asENExposureConfiguration))
+      var newExposures: [Exposure] = []
+      if let summary = exposureSummary {
```

Comment:
Note that a difference here is that the OS will now keep track of older exposures for you.
What happens if, yesterday, the app detected an exposure that happened a few days ago, and now you run this code again? We wouldn't want to notify again for an exposure we already knew about.

Reply:
uhmmm yes, should would filter exposures by date before scoring?

Reply:
I'd want to hear what @mattThousand has to say here as well, but here's what I would think:
- store exposures that led to notifications. this should be at most one per day, since we now have daily summaries.
  I think we do this already right, for the calendar? With the code in Exposure.swift above?
- for each newExposures, that happened on a day that we did *not* already notify for, do notify again
- save new exposures that led to notifications
- It looks to me as if we currently only save the date as meta data for the exposure? If so, we're done. If we care about duration as well, or other meta data, we should make sure to potentially update existing exposures as well, in case we added more exposure to a day we already notified for.

Reply:
If a day should have at most one exposure, it sounds like you will want to store any exposures that result in notifications in realm, and each time this codepath is run, filter resulting exposures for dates that already have exposures in realm